charts/stable/synapse/templates/_configmap.tpl

```diff
@@ -86,6 +86,26 @@ data:
 
     enable_registration: {{ .Values.matrix.registration.enabled }}
 
+    {{- if .Values.matrix.registration.enabled }}
+    registration_requires_token: {{ .Values.matrix.registration.requiresToken }}
+
+    {{- if .Values.matrix.registration.require3PID }}
+    registrations_require_3pid:
+        {{- range .Values.matrix.registration.requires3PID }}
+        - {{ . | .medium }}
+        {{- end }}
+    {{- end }}
+
+    disable_msisdn_registration: {{ .Values.matrix.registration.disableMSISDNRegistration }}
+    enable_3pid_lookup: {{ .Values.matrix.registration.enable3PIDLookup }}
+
+    {{- if .Values.matrix.registration.allowedLocal3PIDs }}
+    allowed_local_3pids:
+        {{ .Values.matrix.registration.allowedLocal3PIDs }}
```

Comment:
This is outputting `[map[medium:email pattern:.+@not-gmail.com]]`

Reply:
You should insert using a toYaml thingy... I suspect